great_expectations/rule_based_profiler/profiler.py

```diff
@@ -98,8 +103,11 @@ class Profiler:
     def __init__(
         self,
         *,
-        profiler_config: Optional[Dict[str, Dict[str, Dict[str, Any]]]] = None,
-        data_context: Optional[DataContext] = None,
+        name: str,
+        config_version: float,
+        rules: Dict[str, RuleConfig],
+        variables: Optional[Dict[str, Any]] = None,
+        data_context: Optional["DataContext"] = None,
```

Comment:
Separate out `profiler_config` into it's core objects (the ones defined in the schema)